tezos/context/src/tezedge_context.rs

```diff
@@ -70,98 +72,125 @@ impl TezedgeIndex {
         }
     }
 
-    pub fn find_entry_bytes(&self, hash: HashId) -> Result<Option<Vec<u8>>, DBError> {
+    /// Fetches data from the repository associated to this `hash_id`.
+    ///
+    /// This returns the raw owned value (`Vec<u8>`).
+    /// `Self::find_object` should be used to avoid allocating a `Vec<u8>`.
+    pub fn find_object_bytes(&self, hash_id: HashId) -> Result<Option<Vec<u8>>, DBError> {
         let repo = self.repository.read()?;
-        Ok(repo.get_value(hash)?.map(|v| v.to_vec()))
+        Ok(repo.get_value(hash_id)?.map(|v| v.to_vec()))
     }
 
-    /// Find data from the repository and deserialize it into `storage`
-    pub fn find_entry(
+    /// Fetches data from the repository and deserialize it into `Self::storage`.
+    ///
+    /// Returns the deserialized `Object`.
+    pub fn find_object(
         &self,
-        hash: HashId,
+        hash_id: HashId,
         storage: &mut Storage,
-    ) -> Result<Option<Entry>, DBError> {
-        match self.find_entry_bytes(hash)? {
+    ) -> Result<Option<Object>, DBError> {
+        let repo = self.repository.read()?;
+
+        match repo.get_value(hash_id)? {
             None => Ok(None),
-            Some(entry_bytes) => {
-                let repository = self.repository.read()?;
-                Ok(Some(deserialize(
-                    entry_bytes.as_ref(),
-                    storage,
-                    &*repository,
-                )?))
-            }
+            Some(object_bytes) => Ok(Some(deserialize(object_bytes.as_ref(), storage, &*repo)?)),
         }
     }
 
-    pub fn get_hash(&self, hash_id: HashId) -> Result<Option<EntryHash>, DBError> {
+    /// Returns the raw `ObjectHash` associated to this `hash_id`.
+    ///
+    /// It reads it from the repository.
+    pub fn get_hash(&self, hash_id: HashId) -> Result<Option<ObjectHash>, DBError> {
         Ok(self
             .repository
             .read()?
             .get_hash(hash_id)?
             .map(|h| h.into_owned()))
     }
 
-    pub fn get_entry(&self, hash: HashId, storage: &mut Storage) -> Result<Entry, MerkleError> {
-        match self.find_entry(hash, storage)? {
-            None => Err(MerkleError::EntryNotFound { hash_id: hash }),
-            Some(entry) => Ok(entry),
+    /// Fetches data from the repository and deserialize it into `storage`.
+    ///
+    /// Returns an error when the object was not found.
+    pub fn get_object(&self, hash: HashId, storage: &mut Storage) -> Result<Object, MerkleError> {
```

Comment:
"Fetches an object from the repository..." (I think we should explicitly say "object" instead of "data" when it is actual "objects" we are working with).

Also, would `fetch_object` be a better name here?